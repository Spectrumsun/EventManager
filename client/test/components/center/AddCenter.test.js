import React from 'react';
import { mount, shallow } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { createMemoryHistory } from 'history';
import render from 'react-test-renderer';
import thunk from 'redux-thunk';
import sinon from 'sinon';
import ConnectedAddCenter,
{ AddCenter } from '../../../src/components/Center/AddCenter';
import CenterFrom from '../../../src/components/Center/Form/CenterForm';


const middleware = [thunk];
const mockStore = configureStore(middleware);
const initialState = {
  centers: {
    pagination: {
      message: 'success',
      result: [
        {
          id: 29,
          centerName: 'farm center',
          serId: 1
        }
      ],
      count: 10,
      pages: 2
    }
  }
};

const store = mockStore(initialState);

const props = {
  initPostCenters: sinon.spy(() => new Promise((cb) => {
    cb();
  })),
  history: createMemoryHistory()
};

const mountedWrapper = mount(  
  <Provider store={store}>
    <BrowserRouter>
      <ConnectedAddCenter {...props} />
    </BrowserRouter>
  </Provider>);

const shallowWrapper = shallow(<AddCenter {...props} />);

const state = {
  name: 'yaba',
  city: 'lagos',
  address: 'nO 22 yaba',
  about: 'lol',
  availability: 'yes',
  values: 'table',
  facility: ['car, egg'],
  image: 'http://wwww.image.com/png',
  preview: 'folder/image',
  imageurl: 'http://wwww.image.com/png',
  publicUrlId: 'png',
  progress: `${0}%`,
  formValid: false,
};


const event = {
  preventDefault: jest.fn(),
  target: {
    name: 'yaba',
    city: 'lagos',
    address: 'nO 22 yaba',
    about: 'lol',
    availability: 'yes',
    values: 'table',
    facility: ['car, egg'],
    image: 'erere',
    preview: 'folder/image',
    imageurl: 'http://wwww.image.com/png',
    publicUrlId: 'png',
    progress: `${0}%`,
    formValid: false,
    files: '3'
  }
};

const i = [1, 2, 3, 4]

// let wrapper;

describe('<AddCenter /> Component', () => {
  const wrapper = shallow(<AddCenter {...props} />);
  it('should render the <AddCenter />', () => {
    shallow(<AddCenter {...props} />);
  });

  it('should render the <AddCenter /> without crashing', () => {
    expect(mountedWrapper).toBeDefined();
    expect(mountedWrapper.find('AddCenter').length).toBe(1);
  });

  it('should match component snapshot', () => {
    const tree = render.create(<Provider store={store}>
      <BrowserRouter>
        <ConnectedAddCenter {...props} />
      </BrowserRouter>
    </Provider>);
    expect(tree).toMatchSnapshot();
  });


  it('should render initial layout of <AddCenter />', () => {
    expect(wrapper.getElements()).toMatchSnapshot();
  });

  it('should have three div on layout', () => {
    expect(wrapper.find('div').length).toEqual(4);
  });

  it('should have <CenterFrom /> when the page loads', () => {
    expect(wrapper.find(CenterFrom)).toHaveLength(1);
  });

  it('calls onChange event', () => {
    sinon.spy(shallowWrapper.instance(), 'onChange');
    shallowWrapper.instance().onChange(event);
    expect(shallowWrapper.instance().onChange.calledOnce).toEqual(true);
  });

  it('calls onSubmit event', () => {
    sinon.spy(shallowWrapper.instance(), 'onSubmit');
    shallowWrapper.setState(state);
    shallowWrapper.instance().onSubmit(event);
    expect(shallowWrapper.instance().onSubmit.calledOnce).toEqual(true);
  });

  it('calls onClick event', () => {
    sinon.spy(shallowWrapper.instance(), 'onClick');
    shallowWrapper.setState(state);
    shallowWrapper.instance().onClick();
    expect(shallowWrapper.instance().onClick.calledOnce).toEqual(true);
  });

  it('calls onKeyPress event', () => {
    sinon.spy(shallowWrapper.instance(), 'onKeyPress');
    shallowWrapper.setState(state);
    shallowWrapper.instance().onKeyPress(event);
    expect(shallowWrapper.instance().onKeyPress.calledOnce).toEqual(true);
  });

  it('calls removeFacility event', () => {
    sinon.spy(shallowWrapper.instance(), 'removeFacility');
    shallowWrapper.setState(state);
    shallowWrapper.instance().removeFacility(i);
    expect(shallowWrapper.instance().removeFacility.calledOnce).toEqual(true);
  });

  it('should have three div element match snap', () => {
    expect(wrapper.getElements('div')).toMatchSnapshot();
  });


  it('should have div element', () => {
    expect(wrapper.find('div').length).toEqual(4);
  });
});

