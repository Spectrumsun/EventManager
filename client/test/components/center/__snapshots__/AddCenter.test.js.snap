// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddCenter /> Component should have three div element match snap 1`] = `
Array [
  <div>
    <div
      className="container"
      style={
        Object {
          "paddingTop": "100px",
        }
      }
    >
      <div
        className="card card w-50 loginCard "
      >
        <div
          className="card-header dark"
        >
          <h1
            className="color"
          >
            Add Center
          </h1>
        </div>
        <centerFrom
          about=""
          address=""
          availability=""
          city=""
          disabled=""
          facility={Array []}
          formValid={false}
          handleImageChange={[Function]}
          imagePreview={null}
          name=""
          onChange={[Function]}
          onClick={[Function]}
          onKeyPress={[Function]}
          onSubmit={[Function]}
          progress="0%"
          removeFacility={[Function]}
          values=""
        />
      </div>
    </div>
  </div>,
]
`;

exports[`<AddCenter /> Component should match component snapshot 1`] = `
<div>
  <div
    className="container"
    style={
      Object {
        "paddingTop": "100px",
      }
    }
  >
    <div
      className="card card w-50 loginCard "
    >
      <div
        className="card-header dark"
      >
        <h1
          className="color"
        >
          Add Center
        </h1>
      </div>
      <div
        className="card-body"
      >
        <form
          onKeyPress={[Function]}
          onSubmit={[Function]}
        >
          <div
            className="form-row"
          >
            <div
              className="form-group col-md-12"
            >
              <h5>
                <label
                  htmlFor="inputEmail4"
                >
                  Center Name
                </label>
              </h5>
              <input
                className="form-control form-control-lg"
                name="name"
                onChange={[Function]}
                required={true}
                type="text"
                value=""
              />
            </div>
            <div
              className="form-group col-md-6"
            >
              <h5>
                <label
                  htmlFor="inputPassword4"
                >
                  City
                </label>
              </h5>
              <input
                className="form-control form-control-lg"
                name="city"
                onChange={[Function]}
                required={true}
                type="text"
                value=""
              />
            </div>
            <div
              className="form-group col-md-6"
            >
              <h5>
                <label
                  htmlFor="inputPassword4"
                >
                  Availability
                </label>
              </h5>
              <input
                className="form-control form-control-lg"
                name="availability"
                onChange={[Function]}
                required={true}
                type="text"
                value=""
              />
            </div>
            <div
              className="form-group col-md-12"
            >
              <h5>
                <label
                  htmlFor="inputPassword4"
                >
                  Address
                </label>
              </h5>
              <input
                className="form-control form-control-lg"
                name="address"
                onChange={[Function]}
                required={true}
                type="text"
                value=""
              />
            </div>
            <div
              className="form-group col-md-12"
            >
              <h5>
                <label
                  htmlFor="inputPassword4"
                >
                  About
                </label>
              </h5>
              <textarea
                className="form-control form-control-lg"
                id="exampleFormControlTextarea1"
                name="about"
                onChange={[Function]}
                required={true}
                rows="5"
                type="text"
                value=""
              />
            </div>
            <div
              className="form-group col-md-12"
            >
              <h5>
                <label
                  htmlFor="inputPassword4"
                >
                  Add Image
                </label>
              </h5>
              <input
                accept="image/gif, image/png, image/jpeg"
                className="btn btn-light"
                name="image"
                onChange={[Function]}
                type="file"
              />
            </div>
            <div
              className="imgPreview"
            />
          </div>
          <h5>
            Add Facilities
          </h5>
          <input
            className="form-control col-md-6 form-control-lg"
            name="values"
            onChange={[Function]}
            style={
              Object {
                "float": "left",
              }
            }
            type="text"
            value=""
          />
          <input
            className="btn btn-info btn-lg"
            disabled={true}
            id="add"
            onClick={[Function]}
            style={
              Object {
                "marginLeft": "20px",
              }
            }
            type="button"
            value="add"
          />
          <br />
          <br />
          <ul
            className="list-group col-md-6"
          />
          <br />
          <button
            className="btn btn-primary btn-lg"
            disabled={false}
            type="submit"
          >
            Submit
          </button>
          <br />
          <br />
          <div
            className="progress"
          >
            <div
              aria-valuemax="100"
              aria-valuemin="0"
              aria-valuenow="0%"
              className="progress-bar progress-bar-striped"
              role="progressbar"
              style={
                Object {
                  "width": "0%",
                }
              }
            >
              0%
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`<AddCenter /> Component should render initial layout of <AddCenter /> 1`] = `
Array [
  <div>
    <div
      className="container"
      style={
        Object {
          "paddingTop": "100px",
        }
      }
    >
      <div
        className="card card w-50 loginCard "
      >
        <div
          className="card-header dark"
        >
          <h1
            className="color"
          >
            Add Center
          </h1>
        </div>
        <centerFrom
          about=""
          address=""
          availability=""
          city=""
          disabled=""
          facility={Array []}
          formValid={false}
          handleImageChange={[Function]}
          imagePreview={null}
          name=""
          onChange={[Function]}
          onClick={[Function]}
          onKeyPress={[Function]}
          onSubmit={[Function]}
          progress="0%"
          removeFacility={[Function]}
          values=""
        />
      </div>
    </div>
  </div>,
]
`;
